btcec/schnorr/musig2/musig2_test.go

```diff
@@ -588,6 +598,238 @@ func TestMuSig2SigningTestVectors(t *testing.T) {
+			nonces: [][66]byte{
+				verifyPnonce2,
+				verifyPnonce3,
+				verifyPnonce1,
+			},
+			pubnonceIndex: 2,
+			partialSig:    mustParseHex("8C4473C6A382BD3C4AD7BE59818DA5ED7CF8CEC4BC21996CFDA08BB4316B8BC7"),
+			tweaks: []KeyTweakDesc{
+				genTweakParity(tweak1, true),
+				genTweakParity(tweak2, false),
+				genTweakParity(tweak3, true),
+				genTweakParity(tweak4, false),
+			},
+		},
+		// Vector 8
+		{
+
+			partialSig:    signExpected1,
+			pubnonceIndex: 0,
+			keyOrder:      []int{0, 1, 2},
+			nonces: [][66]byte{
+				verifyPnonce1,
+				verifyPnonce2,
+				verifyPnonce3,
+			},
+		},
+		// Vector 9
+		{
+
+			partialSig:    signExpected2,
+			pubnonceIndex: 1,
+			keyOrder:      []int{1, 0, 2},
+			nonces: [][66]byte{
+				verifyPnonce2,
+				verifyPnonce1,
+				verifyPnonce3,
+			},
+		},
+		// Vector 10
+		{
+
+			partialSig:    signExpected3,
+			pubnonceIndex: 2,
+			keyOrder:      []int{1, 2, 0},
+			nonces: [][66]byte{
+				verifyPnonce2,
+				verifyPnonce3,
+				verifyPnonce1,
+			},
+		},
+		// Vector 11: Wrong signature (which is equal to the negation of valid signature expected[0])
+		{
+
+			partialSig:    mustParseHex("97AC833ADCB1AFA42EBF9E0725616F3C9A0D5B614F6FE283CEAAA37A8FFAF406"),
+			pubnonceIndex: 0,
+			keyOrder:      []int{0, 1, 2},
+			nonces: [][66]byte{
+				verifyPnonce1,
+				verifyPnonce2,
+				verifyPnonce3,
+			},
+			expectedError: ErrPartialSigInvalid,
+		},
+		// Vector 12: Wrong signer
+		{
+
+			partialSig:    signExpected1,
+			pubnonceIndex: 1,
+			keyOrder:      []int{0, 1, 2},
+			nonces: [][66]byte{
+				verifyPnonce1,
+				verifyPnonce2,
+				verifyPnonce3,
+			},
+			expectedError: ErrPartialSigInvalid,
+		},
+		// Vector 13: Signature exceeds group size
+		{
+
+			partialSig:    mustParseHex("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141"),
+			pubnonceIndex: 0,
+			keyOrder:      []int{0, 1, 2},
+			nonces: [][66]byte{
+				verifyPnonce1,
+				verifyPnonce2,
+				verifyPnonce3,
+			},
+			expectedError: ErrPartialSigInvalid,
+		},
+		// Vector 14: Invalid pubnonce
+		{
+
+			partialSig:    signExpected1,
+			pubnonceIndex: 0,
+			keyOrder:      []int{0, 1, 2},
+			nonces: [][66]byte{
+				canParsePubNonce("020000000000000000000000000000000000000000000000000000000000000009"),
+				verifyPnonce2,
+				verifyPnonce3,
+			},
+			expectedError: secp256k1.ErrPubKeyNotOnCurve,
+		},
+		// Vector 15: Invalid public key
+		{
+
+			partialSig:    signExpected1,
+			pubnonceIndex: 0,
+			keyOrder:      []int{3, 1, 2},
+			nonces: [][66]byte{
+				verifyPnonce1,
+				verifyPnonce2,
+				verifyPnonce3,
+			},
+			expectedError: secp256k1.ErrPubKeyNotOnCurve,
+		},
+	}
+
+	for _, testCase := range testCases {
+
+		// todo find name
+		testName := fmt.Sprintf("%v/tweak=%v", testCase.pubnonceIndex, testCase.keyOrder)
+
+		t.Run(testName, func(t *testing.T) {
+
+			combinedNonce, err := AggregateNonces(testCase.nonces)
+			if err != nil {
+				if errors.Is(err, testCase.expectedError) {
```

Comment:
Similar comment re using a `switch` statement here to make the error handling more readable.